src/rules/no_curly_quotes.rs

```diff
@@ -0,0 +1,110 @@
+use std::fs::File;
+use std::io::BufRead;
+use std::io::BufReader;
+use std::path::PathBuf;
+
+use crate::run::Run;
+
+use anyhow::Context;
+use log::debug;
+use log::trace;
+use rayon::iter::{IntoParallelRefIterator, ParallelIterator};
+
+use crate::diagnostic::{Diagnostic, Position, Range, Replacement, Severity};
+
+pub fn no_curly_quotes(run: &Run, _upstream: &str) -> anyhow::Result<Vec<Diagnostic>> {
+    let config = &run.config.nocurlyquotes;
+
+    if !config.enabled {
+        trace!("'nocurlyquotes' is disabled");
+        return Ok(vec![]);
+    }
+
+    if run.is_upstream() {
+        return Ok(vec![]);
+    }
```

Comment:
What's wrong with running this on the upstream and having it behave like a normal linter? This shouldn't need upstream awareness like ICTC does...

Reply:
I had disabled that for testing - good catch
shoudve put in a donotand